### ISSUE TITLE
test suites: `results.csh` miscounts number of run failures

### ISSUE THREAD

**Issue by phil-blain** [CICE-Consortium/CICE#757]
The`results.csh`  script greps for "`run `" but some tests (`smoke` for example) writes just "`run`" with no trailing space, so the count is off. A quick fix is removing the trailing space  from the grep calls:

```diff
diff --git a/./configuration/scripts/tests/cice.results.csh b/./configuration/scripts/tests/cice.results.csh
index 13580c0..c320167 100644
--- a/./configuration/scripts/tests/cice.results.csh
+++ b/./configuration/scripts/tests/cice.results.csh
@@ -5,9 +5,9 @@ cat ./results.log | grep -iv "#machinfo" | grep -iv "#envinfo"
 set pends     = `cat ./results.log | grep PEND | wc -l`
 set misses    = `cat ./results.log | grep MISS | wc -l`
 set failures  = `cat ./results.log | grep FAIL | wc -l`
-set failbuild = `cat ./results.log | grep FAIL | grep " build " | wc -l`
-set failrun   = `cat ./results.log | grep FAIL | grep " run " | wc -l`
-set failtest  = `cat ./results.log | grep FAIL | grep " test " | wc -l`
+set failbuild = `cat ./results.log | grep FAIL | grep " build" | wc -l`
+set failrun   = `cat ./results.log | grep FAIL | grep " run" | wc -l`
+set failtest  = `cat ./results.log | grep FAIL | grep " test" | wc -l`
 set failcomp  = `cat ./results.log | grep FAIL | grep " compare " | wc -l`
 set failbfbc  = `cat ./results.log | grep FAIL | grep " bfbcomp " | wc -l`
 set failgen   = `cat ./results.log | grep FAIL | grep " generate " | wc -l`
```

the `compare`, `bfbcomp` and `generate` items are always followed by a space and another word, so they can keep their trailing space.

Or, the individual tests could be reviewed and a space added after `run` when it is written to the `test_output`. I think the above is easier but I wanted to check what others think. 

@apcraig

**Comment by phil-blain** [comment]
Another related thing I just noticed: the `complog` failures are not individually listed at the bottom of the `results.log` file (but they are correctly counted in the total failures).

**Comment by phil-blain** [comment]
And here's a tangentially related  patch I've been keeping locally for a while: 7a4eb47cae86fe4fec6a6f65f8fe10ccfd201057
(https://github.com/CICE-Consortium/CICE/compare/main...phil-blain:CICE:suite-results-keep-log)

```diff
commit 4d33e5340bbd9d14ce167234f0245b2c0d204993 (HEAD -> suite-results-keep-log, origin/suite-results-keep-log)
Author: Philippe Blain <philippe.blain@canada.ca>
Date:   Thu Aug 11 16:25

    cice.results.csh: keep a timestampped 'results.log'

diff --git a/configuration/scripts/tests/cice.results.csh b/configuration/scripts/tests/cice.results.csh
index 13580c0..086364a 100644
--- a/configuration/scripts/tests/cice.results.csh
+++ b/configuration/scripts/tests/cice.results.csh
@@ -32,6 +32,9 @@ echo "    #failcomp  = $failcomp" >> results.log
 echo "    #failbfbc  = $failbfbc" >> results.log
 echo "    #failgen   = $failgen" >> results.log

+set stamp = `date '+%y%m%d-%H%M%S'`
+cp results.log results.$stamp.log
+
 echo ""
 echo "Descriptors:"
 echo " PASS - successful completion"
```

I find this useful when re-running suites, to keep a log (no pun intended) of which tests failed in the original run of the suite versus which then passed in subsequent re-runs.

It would be nice if you cherry-picked that patch :) I don't care too deeply about the name of the timestamped log, it could as well be `results.log$stamp` instead of `results.$stamp.log`, or something else - as long as it's time-stamped.

**Comment by apcraig** [comment]
Closed by #772
